odoo/addons/base/tests/test_res_users.py

```diff
@@ -217,3 +308,85 @@ def test_reified_groups_on_change(self):
 
         setattr(user_form, group_field_name, group_public.id)
         self.assertTrue(user_form.share, 'The groups_id onchange should have been triggered')
+
+    def test_user_group_inheritance_no_warning(self):
+        """
+            The warning should not be there when the User tries to Change to Higher rights.
+        """
+        with Form(self.test_group_user, view='base.view_users_form') as UserForm:
+            UserForm._values[self.sales_cat_field] = self.group_sales_administrator.id
+            UserForm._perform_onchange([self.sales_cat_field])
+
+            self.assertFalse(UserForm.user_group_warning)
+
+    def test_user_group_parent_inheritance_no_warning(self):
+        """
+            The warning should not be there when the User Changes Parent rights to Lower one
+        """
+        with Form(self.test_group_user, view='base.view_users_form') as UserForm:
+            UserForm._values[self.field_service_cat_field] = self.group_field_service_user.id
+            UserForm._perform_onchange([self.field_service_cat_field])
+
+            self.assertFalse(UserForm.user_group_warning)
+
+    def test_user_group_inheritance_warning(self):
```

Comment:
because the method is quite complex from a SQL pov, it would be nice to add a performance test as well imo